iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -622,13 +622,15 @@ void PortManager::sendToAllMatchingSubscriberPorts(const capro::CaproMessage& me
 bool PortManager::isCompatibleClientServer(const popo::ServerPortRouDi& server,
                                            const popo::ClientPortRouDi& client) const noexcept
 {
-    auto requestMatch = !(client.getServerTooSlowPolicy() == popo::ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA
-                          && server.getRequestQueueFullPolicy() == popo::QueueFullPolicy::BLOCK_PRODUCER);
+    const auto serviceMatch = client.getCaProServiceDescription() == server.getCaProServiceDescription();
 
-    auto responseMatch = !(server.getClientTooSlowPolicy() == popo::ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA
-                           && client.getResponseQueueFullPolicy() == popo::QueueFullPolicy::BLOCK_PRODUCER);
+    const auto requestMatch = !(client.getServerTooSlowPolicy() == popo::ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA
+                                && server.getRequestQueueFullPolicy() == popo::QueueFullPolicy::BLOCK_PRODUCER);
 
-    return requestMatch && responseMatch;
+    const auto responseMatch = !(server.getClientTooSlowPolicy() == popo::ConsumerTooSlowPolicy::DISCARD_OLDEST_DATA
+                                 && client.getResponseQueueFullPolicy() == popo::QueueFullPolicy::BLOCK_PRODUCER);
+
+    return serviceMatch && requestMatch && responseMatch;
```

Comment:
Now I remember why I didn't do the service description check in this method. It does unnecessary checks for all ports even if the service descriptions do not match, which happens most of the time. Could you please do an early return when `serviceMatch` is false? Can you do the same for pub-sub?

Reply:
Done